rebac-admin-backend/v1/identities.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	r "github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// GetIdentities Get list of identities.
+// (GET /identities)
+func (h handler) GetIdentities(w http.ResponseWriter, req *http.Request, params r.GetIdentitiesParams) {
+	identities, err := h.Identities.ListIdentities(&params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := r.GetIdentitiesResponse{
+		Data:   identities.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PostIdentities Add an identity.
+// (POST /identities)
+func (h handler) PostIdentities(w http.ResponseWriter, req *http.Request) {
+	identity := new(r.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, err)
```

Comment:
see.. this is wrong.. writeErrorResponse will always return an internal server error status.. but in this case it's more of a "bad request", since we cannot decode it

Reply:
my bad, I'll fix this by passing a validation error which will translate to a bad request, thanks for spotting it!